examples/jsm/loaders/PLYLoader.js

```diff
@@ -589,17 +605,62 @@ class PLYLoader extends Loader {
 
 		}
 
+		function extractHeaderText( bytes ) {
+
+			let i = 0;
+			let cont = true;
+
+			let line = '';
+			const lines = [];
+
+			do {
+
+				const c = String.fromCharCode( bytes[ i++ ] );
+
+				if ( c !== "\n" && c !== "\r" ) {
+
+					line += c;
+
+				} else {
+
+					if ( line === 'end_header' ) cont = false;
+					if ( line !== '' ) {
+
+						lines.push( line );
+						line = '';
+
+					}
+
+				}
+
+			}  while ( cont && i < bytes.length );
```

Comment:
Just for my understanding: There is no way around the do/while loop since we can't extract the header size in advance, right? You have to explicitly look for `end_header` so you know when the header content ends.

I was hoping we could create a smaller array buffer just with the header data and then use `LoaderUtils.decodeText()` again. But if we don't know how big the header is without actually reading and evaluating the data, we can't use this approach.

Reply:
Pretty much.  You could use a guessed value for the maximum probably size of header but I can't see a way of guaranteeing it is accurate, the header could include a lot of comment lines for example.

Reply:
Off topic, is it possible to deprecate LoaderUtils.decodeText() now IE11 is not supported, and use TextDecoder()?

Reply:
> Off topic, is it possible to deprecate LoaderUtils.decodeText() now IE11 is not supported, and use TextDecoder()?

AFAICS, all browser supporting ECMAScript 2018 should support `TextDecoder`. So I guess we can do this, yes.

Reply:
Although I'm not sure about node environments. If there is no `TextDecoder` available, `LoaderUtils.decodeText()` still does work because of its fallback.

Reply:
I hadn't thought about node, however, it looks like it has been available for awhile now.

https://nodejs.org/api/util.html#class-utiltextdecoder